src/conventional/sonde_bufr2ioda.py

```diff
@@ -0,0 +1,665 @@
+        elif len(meta_data['latDisplacement']) > 0:
+            target_number = len(meta_data['latDisplacement'])
+        elif len(meta_data['latitude']) > 0:
+            target_number = len(meta_data['latitude'])
+    if target_number == 0:
+        print("HOW on earth is target_number zero?  BUFR sucks!")
+        return data, count, start_pos
+
+    count[1] += target_number
+
+    # For any of the MetaData elements that were totally lacking, fill entire vector with missing.
+    empty = []
+    for k, v in metaDataKeyList.items():
+        if not any(meta_data[k]):
+            meta_data[k] = assign_missing_meta(empty, k, target_number, 0)
+        length = len(meta_data[k])
+        if (length == 1):
+            meta_data[k] = np.full(target_number, meta_data[k][0])
+        elif (length < target_number):
+            logging.warning(f"Key called {k} contains only {length} "
+                            f" elements, whereas {target_number} were expected.")
+            meta_data[k] = assign_missing_meta(meta_data[k], k, target_number, length-1)
+        elif (length > target_number):
+            logging.warning(f"Key called {k} contains {length} "
+                            f" elements, whereas {target_number} were expected.")
+            meta_data[k] = meta_data[k][:target_number]
+
+    # Sondes are special with a launch time and time displacement.
+    if meta_data['timeDisplacement'].any():
+        meta_data['dateTime'] = specialty_time(bufr, meta_data['timeDisplacement'])
+        meta_data['releaseTime'] = np.full(target_number, meta_data['dateTime'][0])
+    else:
+        count[2] += target_number
+        return data, count, start_pos
+
+    # Sondes also have lat/lon displacement from launch/release location.
+    if (meta_data['latDisplacement'].any() and meta_data['lonDisplacement'].any()):
+        for n, delta_lat in enumerate(meta_data['latDisplacement']):
+            delta_lon = meta_data['lonDisplacement'][n]
+            meta_data['latitude'][n] = meta_data['latitude'][0] + delta_lat
+            meta_data['longitude'][n] = meta_data['longitude'][0] + delta_lon
+    else:
+        count[2] += target_number
+        return data, count, start_pos
+
+    # Force longitude into space of -180 to +180 only. Reset both lat/lon missing if either absent.
+    mask_lat = np.logical_or(meta_data['latitude'] < -90.0, meta_data['latitude'] > 90.0)
+    mask_lon = np.logical_or(meta_data['longitude'] < -180.0, meta_data['longitude'] > 360.0)
+    meta_data['latitude'][mask_lat] = float_missing_value
+    meta_data['longitude'][mask_lon] = float_missing_value
+    meta_data['latitude'][mask_lon] = float_missing_value
+    meta_data['longitude'][mask_lat] = float_missing_value
+    for n, longitude in enumerate(meta_data['longitude']):
+        if (meta_data['longitude'][n] != float_missing_value and meta_data['longitude'][n] > 360):
+            meta_data['longitude'][n] = 360.0 - meta_data['longitude'][n]
```

Comment:
My apologies if I'm misreading/misinterpreting.

It looks like the code on lines 534-538 is going to put in a missing value if the longitude is < -180 or > 360 (ie the longitude is out of range for either 0-360 or +/- 180 schemes).

If I've got that right, then Is the code on lines 540-541 meant to translate the 0-360 scheme to the +/- 180 scheme?

If so, then shouldn't the logic be like this:
```
# Test if longitude is greater than 180 and if so subtract 360 to shift to +/- 180 value
if (meta_data['longitude'][n] != float_missing_value and meta_data['longitude'][n] > 180.0):
            meta_data['longitude'][n] = meta_data['longitude'][n] - 360.0
```
Eg, if longitude is 270, then the current code leaves it as 270, and the suggested code would change it to -90; while for either code if the longitude is 90, then it remains 90.

Reply:
Thank you Steve for being thorough.  You are absolutely correct and I made an error here for certain.  Now correcting it.